BootloaderCorePkg/Library/BootloaderCoreLib/BootloaderCoreLib.c

```diff
@@ -444,3 +444,72 @@ GetTempRamInfo (
 
   return EFI_SUCCESS;
 }
+
+/**
+  This function retrieves the address of S3Data.
+
+  @retval    S3Data pointer address.
+
+**/
+VOID *
+EFIAPI
+GetS3DataPtr (
+  VOID
+  )
+{
+  return GetLoaderGlobalDataPointer()->S3DataPtr;
+}
+
+/**
+  This function sets features configuration.
+
+  @param Features   The features the platform supports.
+
+**/
+VOID
+EFIAPI
+SetFeatureCfg (
+  IN  UINT32  Features
+  )
+{
+  GetLoaderGlobalDataPointer()->LdrFeatures = Features;

```

Comment:
Better to OR a specific feature than assignment?

Reply:
We already have GetFeatureCfg(). May add a new API FeatureCfgAndThenOr (UINT32  AndData,  UINT32  OrData)?
Agree it could simplify the caller.

Reply:
This is to make CorePkg APIs simple with primitive Get()/Set() API pair. Using Get()/Set() will minimize changes from original code. Let me keep this implementation to avoid unexpected issues with AndThenOr(). If AndThenOr() is required, we can add it later.

```
From:
  LdrGlobal = (LOADER_GLOBAL_DATA *)GetLoaderGlobalDataPointer ();
  LdrGlobal->LdrFeatures |= FeaturePcdGet (PcdAcpiEnabled)?FEATURE_ACPI:0;
  LdrGlobal->LdrFeatures |= FeaturePcdGet (PcdVerifiedBootEnabled)?FEATURE_VERIFIED_BOOT:0;
  LdrGlobal->LdrFeatures |= FeaturePcdGet (PcdMeasuredBootEnabled)?FEATURE_MEASURED_BOOT:0;
  LdrGlobal->LdrFeatures &= ~FEATURE_ACPI;
  LdrGlobal->LdrFeatures &= ~FEATURE_MEASURED_BOOT;
  LdrGlobal->LdrFeatures &= ~FEATURE_VERIFIED_BOOT;
To:
  LdrFeatures  = GetFeatureCfg ();
  LdrFeatures |= FeaturePcdGet (PcdAcpiEnabled)?FEATURE_ACPI:0;
  LdrFeatures |= FeaturePcdGet (PcdVerifiedBootEnabled)?FEATURE_VERIFIED_BOOT:0;
  LdrFeatures |= FeaturePcdGet (PcdMeasuredBootEnabled)?FEATURE_MEASURED_BOOT:0;
  LdrFeatures &= ~FEATURE_ACPI;
  LdrFeatures &= ~FEATURE_MEASURED_BOOT;
  LdrFeatures &= ~FEATURE_VERIFIED_BOOT;
  SetFeatureCfg (LdrFeatures);
```